gax/src/main/java/com/google/api/gax/rpc/ApiCallContext.java

```diff
@@ -126,4 +128,16 @@
    * in the present instance.
    */
   ApiCallContext merge(ApiCallContext inputCallContext);
+
+  /** Return a new ApiCallContext with the extraHeaders merged into the present instance. */
+  @BetaApi("The surface for extra headers is not stable yet and may change in the future.")
+  ApiCallContext withExtraHeaders(Map<String, List<String>> extraHeaders);
+
+  /** Return the extra headers set for this context. */
+  @BetaApi("The surface for extra headers is not stable yet and may change in the future.")
+  Map<String, List<String>> getExtraHeaders();
+
+  /** Return a new ApiCallContext with empty extra headers. */
+  @BetaApi
+  ApiCallContext withEmptyExtraHeaders();
```

Comment:
How is this method different from calling `getExtraHeaders(Collections.emptyMap())`? It is better to avoid introducing new entities (here methods) if not necessary .